src/sessions.rs

```diff
@@ -273,6 +301,69 @@ pub mod user {
     }
 }
 
+
+#[derive(Debug, Deserialize)]
+#[allow(non_snake_case)]
+struct CreateSessionCookieResponseJson {
+    sessionCookie: String,
+}
+
+/// https://cloud.google.com/identity-platform/docs/reference/rest/v1/projects/createSessionCookie
+#[derive(Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[allow(non_snake_case)]
+struct SessionLogin {
+    /// Required. A valid Identity Platform ID token.
+    idToken: String,
+    /// The number of seconds until the session cookie expires. Specify a duration in seconds, between five minutes and fourteen days, inclusively.
+    validDuration: u64,
+}
+
+/// Firebase Auth provides server-side session cookie management for traditional websites that rely on session cookies.
+/// This solution has several advantages over client-side short-lived ID tokens,
+/// which may require a redirect mechanism each time to update the session cookie on expiration:
+///
+/// * Improved security via JWT-based session tokens that can only be generated using authorized service accounts.
+/// * Stateless session cookies that come with all the benefit of using JWTs for authentication.
+///   The session cookie has the same claims (including custom claims) as the ID token, making the same permissions checks enforceable on the session cookies.
+/// * Ability to create session cookies with custom expiration times ranging from 5 minutes to 2 weeks.
+/// * Flexibility to enforce cookie policies based on application requirements: domain, path, secure, httpOnly, etc.
+/// * Ability to revoke session cookies when token theft is suspected using the existing refresh token revocation API.
+/// * Ability to detect session revocation on major account changes.
+///
+/// See https://firebase.google.com/docs/auth/admin/manage-cookies
+///
+/// The generated session cookie is a JWT that includes the firebase user id in the "sub" (subject) field.
+///
+/// Arguments:
+/// - `credentials` The credentials
+/// - `id_token` An access token, sometimes called a firebase id token.
+/// - `duration` The cookie duration
+///
+pub fn create_session_cookie(credentials: &credentials::Credentials, id_token: String, duration: chrono::Duration) -> Result<String, FirebaseError> {
+    // Generate the assertion from the admin credentials
+    let assertion = crate::jwt::session_cookie::create_jwt_encoded(credentials, duration)?;
+
+    // Request Google Oauth2 to retrieve the access token in order to create a session cookie
+    let client = reqwest::blocking::Client::new();
+    let response_oauth2: user::Oauth2Response = client.post(GOOGLE_OAUTH2_URL)
+        .form(&[("grant_type", "urn:ietf:params:oauth:grant-type:jwt-bearer"), ("assertion", &assertion)])
+        .send()?
+        .json()?;
+
+    // Create a session cookie with the access token previously retrieved
+    const CREATE_SESSION_COOKIE_URL: &str = "https://identitytoolkit.googleapis.com/v1/projects/arenel:createSessionCookie";
```

Comment:
I think we need to change the URL. `arenel` is a random project that I created maybe we need to pass the project name as an argument ?